mm2src/trezor/src/transport/udp.rs

```diff
@@ -0,0 +1,161 @@
+/// This source file was borrowed from Trezor repo (https://raw.githubusercontent.com/trezor/trezor-firmware/07ba960ab4aa5aa3ddf16ae74c3658782d491250/rust/trezor-client/src/transport/udp.rs)
+/// and modified to integrate into this project.
+/// Adds udp transport to interact with trezor emulator.
+/// To build emulator use this repo: https://github.com/trezor/trezor-firmware, build with build-docker.sh for the desired branch (tag)
+/// Tested with the legacy emulator (for Trezor One).
+/// After building the emulator find it as ./legacy/firmware/trezor.elf file.
+/// Start it (no params needed) and initialize it with Trezor Suite: create a wallet, find a receive address.
+/// You need the bridge for connecting from the Suite, it can be downloaded from trezor.io.
+/// Do not use pin for the created wallet.
+/// Be aware that when you rebuild the firmware the emulator flash memory file emulator.img is recreated (so save it before rebuilding code)
+use super::{protocol::{Link, Protocol, ProtocolV1},
+            ProtoMessage, Transport};
+use crate::{TrezorError, TrezorResult};
+use async_std::{io, net::UdpSocket, task::block_on};
+use mm2_err_handle::prelude::*;
+use std::time::Duration;
+
+// A collection of constants related to the Emulator Ports.
+mod constants {
+    pub const DEFAULT_HOST: &str = "127.0.0.1";
+    pub const DEFAULT_PORT: &str = "21324";
+    pub const DEFAULT_DEBUG_PORT: &str = "21325";
+    pub const LOCAL_LISTENER: &str = "127.0.0.1:0";
+}
+
+use async_trait::async_trait;
+use constants::{DEFAULT_DEBUG_PORT, DEFAULT_HOST, DEFAULT_PORT, LOCAL_LISTENER};
+
+/// The chunk size for the serial protocol.
+const CHUNK_SIZE: usize = 64;
+
+const READ_TIMEOUT_MS: u64 = 100000;
+const WRITE_TIMEOUT_MS: u64 = 100000;
+
+/// A device found by the `find_devices()` method.  It can be connected to using the `connect()`
+/// method.
+pub struct UdpAvailableDevice {
+    //pub model: Model,
+    pub debug: bool,
+    transport: UdpTransport,
+}
+
+impl UdpAvailableDevice {
+    /// Connect to the device.
+    pub fn connect(self) -> TrezorResult<UdpTransport> {
+        block_on(async {
+            let transport = UdpTransport::connect(&self).await?;
+            Ok(transport)
+        })
+    }
+}
+
+pub fn find_devices() -> TrezorResult<Vec<UdpAvailableDevice>> {
+    block_on(async {
```

Comment:
It's better to make this function async instead of using `block_on`

Reply:
done